client.go

```diff
@@ -0,0 +1,92 @@
+package radius
+
+import "sync"
+
+// Clients are list of client allowed to communicate with server
+type Clients struct {
+	herd map[string]Client
+	sync.RWMutex
+}
+
+// Get client from list of clients based on host
+func (cls *Clients) Get(host string) Client {
+	cls.RLock()
+	defer cls.RUnlock()
+	cl, _ := cls.herd[host]
+	return cl
+}
+
+// Add new client or reset existing client based on host
+func (cls *Clients) AddOrUpdate(cl Client) {
+	cls.Lock()
+	defer cls.Unlock()
+	cls.herd[cl.GetHost()] = cl
+}
+
+// Remove client based on host
+func (cls *Clients) Remove(host string) {
+	cls.Lock()
+	defer cls.Unlock()
+	delete(cls.herd, host)
+}
+
+// SetHerd reset/initialize the herd of clients
+func (cls *Clients) SetHerd(herd []Client) {
+	cls.Lock()
+	defer cls.Unlock()
+	if cls.herd == nil {
+		cls.herd = make(map[string]Client)
+	}
+	for _, v := range herd {
+		cls.herd[v.GetHost()] = v
+	}
+}
+
+func (cls *Clients) GetHerd() []Client {
+	cls.RLock()
+	defer cls.RUnlock()
+	herd := make([]Client, len(cls.herd))
+	i := 0
+	for _, v := range cls.herd {
+		herd[i] = v
+		i++
+	}
+	return herd
+}
+
+// Client represent a client to connect to server
+type Client interface {
```

Comment:
How is this interface can be used?
I think this is too complex for the client implement.
Please keep the code simple and stupid.